3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/CurrencyConversionEnrichmentSpec.scala

```diff
@@ -0,0 +1,86 @@
+      (_, trCurrency, apiKey, trAmountTotal, trAmountTax, trAmountShipping, tiCurrency, tiPrice, dateTime, expected) =>
+        CurrencyConversionEnrichment(apiKey, "EUR", "EOD_PRIOR").convertCurrencies(Option(trCurrency), trAmountTotal, trAmountTax, trAmountShipping, Option(tiCurrency), tiPrice, Option(dateTime)) must_== expected 
+    }
+
+  def e2=
+    "SPEC NAME"                                  || "TRANSACTION CURRENCY"   | "API Key"                             | "Total Amounts"  |"Total Tax"  |"Shipping"   | "TRANSACTION ITEM CURRENCY"     | "TRANSACTION ITEM PRICE"       | "DateTime"         | "CONVERTED TUPLE"                            |
+    "All Field Absent except Date and Currency"  !! ""                       ! appId                                 ! nullValue        ! nullValue   ! nullValue   ! ""                              ! nullValue                      ! coTstamp           ! allAbsent                                    |
+    "All Field Absent"                           !! ""                       ! appId                                 ! nullValue        ! nullValue   ! nullValue   ! ""                              ! nullValue                      ! errorcoTstamp      ! Failure(NonEmptyList("DateTime Missing"))    |
+    "All Field Absent except Currency"           !! "GBP"                    ! appId                                 ! nullValue        ! nullValue   ! nullValue   ! "GBP"                           ! nullValue                      ! errorcoTstamp      ! Failure(NonEmptyList("DateTime Missing"))    |> {
+      (_, trCurrency, apiKey, trAmountTotal, trAmountTax, trAmountShipping, tiCurrency, tiPrice, dateTime, expected) =>
+        CurrencyConversionEnrichment(apiKey, "EUR", "EOD_PRIOR").convertCurrencies(Option(trCurrency), nullValue, nullValue, nullValue, Option(tiCurrency), nullValue, Option(dateTime)) must_== expected    }
+
+  def e3 =
+    "SPEC NAME"                         || "TRANSACTION CURRENCY"   | "API Key"                             | "Total Amounts"  |"Total Tax"  |"Shipping"   | "TRANSACTION ITEM CURRENCY"     | "TRANSACTION ITEM PRICE"                | "CONVERTED TUPLE"                            |
+    "Valid APP ID and API key"          !! "USD"                    ! appId                                 ! 14               ! 4.67        ! 0           ! "GBP"                           ! 10.99                                   ! ("10.12", "3.37", "0.00", "12.74")           |
+    "Valid APP ID and API key"          !! "GBP"                    ! appId                                 ! 16               ! 2.67        ! 0           ! "GBP"                           ! 10.00                                   ! ("18.54", "3.09", "0.00", "11.59")           |> {
```

Comment:
These tests aren't quite right. Remember, that a given enriched event will have either the transaction (tr_) fields set, or the transaction item (ti_) fields set, or neither. Never both.